deserialize.go

```diff
@@ -33,6 +33,28 @@ func DeserializeVerticesFromBytes(rawResponse []byte) ([]Vertex, error) {
 	return response, nil
 }
 
+// DeserializeListFromBytes validates that the provided RawMessage ([]byte) corresponds to a g:List type and returns a slice of values as RawMessages
+func DeserializeListFromBytes(rawResponse json.RawMessage) ([]json.RawMessage, error) {
```

Comment:
Just clarifying why this needs to be a `json.RawMessage` rather than the `[]byte` the other functions take in as an argument. I know it is the same underlying type, but was wondering if there was a specific reason?

Reply:
There is no strong reason. I used json.RawMessage because it was the type that I observed when trying things with the real Neptune connection, and I think it's clearer (more specific than []byte) - but I can change it to []byte for consistency (or change all other methods to use json.RawMessage)

Reply:
No real preference, just thought it was inconsistent. I suppose if I had to make a choice I'd consider the most generic option so in this case I'd go for `[]byte` but I'm happy with whatever you choose.

Reply:
Updated with `[]byte` everywhere, so it's consistent.

Reply:
If we use []byte if fails to deserialize in multiple places, I think we will need to use json.RawMessage - I will update it